src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1559,25 +1557,25 @@ void SolverInterfaceImpl::computeMappings(const utils::ptr_vector<MappingContext
   }
 }
 
-void SolverInterfaceImpl::mapData(DataContext *context, const std::string &mappingType)
+void SolverInterfaceImpl::mapData(DataContext &context, const std::string &mappingType)
 {
   PRECICE_TRACE();
   using namespace mapping;
   MappingConfiguration::Timing timing;
-  if (context->hasMapping()) {
-    timing         = context->mappingContext().timing;
-    bool hasMapped = context->mappingContext().hasMappedData;
+  if (context.hasMapping()) {
+    timing         = context.mappingContext().timing;
+    bool hasMapped = context.mappingContext().hasMappedData;
     bool mapNow    = timing == MappingConfiguration::ON_ADVANCE;
     mapNow |= timing == MappingConfiguration::INITIAL;
     if (mapNow && (not hasMapped)) {
-      int inDataID  = context->getFromDataID();
-      int outDataID = context->getToDataID();
+      int inDataID  = context.getFromDataID();
+      int outDataID = context.getToDataID();
       PRECICE_DEBUG("Map \"{}\" data \"{}\" from mesh \"{}\"",
-                    mappingType, context->getDataName(), context->getMeshName());
-      context->resetToData();
+                    mappingType, context.getDataName(), context.getMeshName());
+      context.resetToData();
```

Comment:
@uekerman Why exactly do we need to reset the to data here. Doesn't the mapping overwrite everything?

Reply:
If I comment out this statement, the tests at least complain. So from that perspective it seems to be necessary. Can we offload this to an issue (if it is an issue) and merge this PR?

Reply:
Some conservative mapping variants are based on an incremental "+=" implementation. There is no clean overwriting. 

Example:

https://github.com/precice/precice/blob/04ee730860b0e7ca3350552593c8d878abf616a0/src/mapping/NearestNeighborMapping.cpp#L65
 
Thus, values need to be set to 0 somewhere.